src/devices/sound/segapcm.cpp

```diff
@@ -20,12 +21,9 @@ segapcm_device::segapcm_device(const machine_config &mconfig, const char *tag, d
 	: device_t(mconfig, SEGAPCM, tag, owner, clock),
 		device_sound_interface(mconfig, *this),
 		device_rom_interface(mconfig, *this, 21),
-		m_ram(nullptr),
-		m_bank(0),
-		m_bankshift(0),
-		m_bankmask(0),
 		m_stream(nullptr)
 {
+	std::fill(std::begin(m_ram), std::end(m_ram), 0);
```

Comment:
Zero-filling `m_ram` on construction is wasted work when it's going to be filled with 0xff at start time.  But that aside, why did you change it from being dynamically allocated to being a member array?  Making it dynamically allocated at start time can improve the performance of `-validate`, `-listxml`, `-romident` and other verbs that need to instantiate devices but not start them.  Making it a member and filling it at construction increases the work required to construct an instance of the device.